src/PrestaShopBundle/Bridge/AdminController/LegacyControllerBridge.php

```diff
@@ -0,0 +1,306 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Bridge\AdminController;
+
+use Context;
+use Hook;
+use Media;
+use Psr\Container\ContainerInterface;
+use Tools;
+
+class LegacyControllerBridge implements LegacyControllerBridgeInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
```

Comment:
It's not a good thing to pass the container to an object, I think you could inject the service you need instead

Reply:
when we move further and copy-paste some more code from legacy, it might be quite annoying to inject a new service every time  $this->container->get('random service') is called, so i thought its ok having whole container. But I can inject single service for now and we'll see how it goes.